service/src/main/java/org/cbioportal/service/impl/CNASegmentServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+/**
+ *
+ * @author jiaojiao
+ */
+package org.cbioportal.service.impl;
+
+import java.util.ArrayList;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import org.cbioportal.model.CNASegmentData;
+import org.cbioportal.model.Gene;
+import org.cbioportal.persistence.CNASegmentRepository;
+import org.cbioportal.persistence.GeneRepository;
+import org.cbioportal.service.CNASegmentService;
+
+@Service
+public class CNASegmentServiceImpl implements CNASegmentService {
+
+    @Autowired
+    private CNASegmentRepository cnaSegmentRepository;
+    
+    @Autowired
+    private GeneRepository geneRepository;
+
+    @Override
+    public List<CNASegmentData> getCNASegmentData(String cancerStudyId, List<String> hugoGeneSymbols, List<String> sampleIds) {
+        List<String> chromosomes = new ArrayList<String>();
```

Comment:
wouldn't we expect duplicates here?  Shouldn't those duplicates be merged into a unique list?  I think Set<String> chromosomes = new Set<String>()  might be what we want.